test/integration/puppetlabs/services/master/environment_transports_int_test.clj

```diff
@@ -0,0 +1,638 @@
+             (is (= mock-response
+                    (response->map test-response-initial))
+                 "unexpected body for test response")
+
+             (str "test data")
+             (str "test data")
+             (purge-all-env-caches)
+
+             (let [production-response-after-all-flush (get-env-transports
+                                                        "production"
+                                                        production-etag-initial)
+                   production-etag-after-all-flush (response-etag
+                                                    production-response-after-all-flush)]
+               (is (= 200 (:status production-response-after-all-flush))
+                   (str
+                    "unexpected status code for prod response after code change "
+                    "and all environment flush"))
+               (is (not (nil? production-etag-after-all-flush))
+                   "no etag returned for production response")
+               (is (not= production-etag-initial production-etag-after-all-flush)
+                   (str
+                    "etag unexpectedly stayed the same even though "
+                    "the production environment changed"))
+               (is (= mock-response
+                      (response->map
+                       production-response-after-all-flush))
+                   "unexpected body for production response"))
+
+             (let [test-response-after-all-flush (get-env-transports
+                                                  "test"
+                                                  test-etag-initial)
+                   test-etag-after-all-flush (response-etag
+                                              test-response-after-all-flush)]
+               (is (= 200 (:status test-response-after-all-flush))
+                   (str
+                    "unexpected status code for test response after code change "
+                    "and all environment flush"))
+               (is (not (nil? test-etag-after-all-flush))
+                   "no etag returned for test response")
+               (is (not= test-etag-initial test-etag-after-all-flush)
+                   (str
+                    "etag unexpectedly stayed the same even though "
+                    "the test environment changed"))
+               (is (= mock-response
+                      (response->map
+                       test-response-after-all-flush))
+                   "unexpected body for test response")))))
+       (catch Exception e
+         (println "dumping puppetserver.log\n" (slurp debug-log))
+         (throw e))))))
+
+(deftest ^:integration
+         not-modified-returned-for-environment-transports-info-request-with-gzip-tag
+  (testing (str "SERVER-1153 - when the webserver gzips the response "
+                "containing environment_transports etag, the next request "
+                "roundtripping that etag returns an HTTP 304 (Not Modified)")
```

Comment:
If I'm reading https://tickets.puppetlabs.com/browse/SERVER-1153 right, it sounds like we may not need to be explicitly stripping `--gzip` anymore, since we're now on Jetty 9.4.x. Could removing [that stripping](https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/services/master/master_core.clj#L244) be part of this, or would that better be done as its own thing in case it's problematic for some reason?

Reply:
I'm for doing it as its own thing and created [SERVER-2534](https://tickets.puppetlabs.com/browse/SERVER-2534)